src/FixedDataTableCellGroup.js

```diff
@@ -57,17 +59,26 @@ var FixedDataTableCellGroupImpl = React.createClass({
   },
 
   componentWillMount() {
+    this._staticCellArray = [];
+    this._columnsToRender = [];
     this._initialRender = true;
   },
 
+  componentWillUnmount() {
+    this._staticCellArray.length = 0;
+    this._columnsToRender.length = 0;
+  },
+
   componentDidMount() {
     this._initialRender = false;
   },
 
   render() /*object*/ {
     var props = this.props;
     var columns = props.columns;
-    var cells = new Array(columns.length);
+
+    this._staticCellArray.length = props.maxVisibleColumns;
+    this._columnsToRender.length = props.maxVisibleColumns;
```

Comment:
I have two re-used arrays. _staticCellArray is the actual cell elements.
_columnsToRender is the array of column indexes that we render. This is essentially the same thing as RowBuffer